server.js

```diff
@@ -1,38 +1,218 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import albums from "./data/albums.json";
+import errorMsg from "./assets/errormsg.json";
+import dotenv from "dotenv";
+
+/*ENVIRONMENT VARIABLES*/
+require("dotenv").config();
+
+/*Database */
+const mongoUrl =
+	process.env.MONGO_URL ||
+	`mongodb+srv://${process.env.DB_USER}:${process.env.DB_KEY}@cluster0.qxpka.mongodb.net/${process.env.DB_NAME}?retryWrites=true&w=majority`;
+
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+/*Model for the album object*/
+const Album = mongoose.model("Album", {
+	number: Number,
+	year: Number,
+	album: String,
+	artist: String,
+	genre: String,
+	subgenre: String,
+});
+
+/*Seed the database if the environment variable reset_tdb is true*/
+if (process.env.RESET_DB) {
+	const seedDatabase = async () => {
+		await Album.deleteMany({});
+		albums.forEach((albumData) => {
+			new Album(albumData).save();
+		});
+	};
+	seedDatabase();
+}
+
+/*API*/
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+const myEndpoints = require("express-list-endpoints");
+
```

Comment:
```suggestion
//_____________Error message in case server is down
app.use((req, res, next) => {
  if(mongoose.connection.readyState === 1) {
    next();
  } else {
    res
    .status(503)
    .send({ error: "service unavailable"});
  };
});
```